### PR TITLE
Onboard SQLServer Flex user resource

### PR DESCRIPTION
**Pull request by GokceGK**

### CODE REVIEW

File: stackit/internal/services/sqlserverflex/user/resource.go [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+		model.ProjectId.ValueString(),
+		model.InstanceId.ValueString(),
+		userId,
+	}
+	model.Id = types.StringValue(
+		strings.Join(idParts, core.Separator),
+	)
+	model.UserId = types.StringValue(userId)
+	model.Username = types.StringPointerValue(user.Username)
+
```
Comment (DiogoFerrao) [inline stackit/internal/services/sqlserverflex/user/resource.go:369]: Is model.Database being mapped?

File: stackit/internal/services/sqlserverflex/user/resource.go [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
```
Comment (DiogoFerrao) [inline stackit/internal/services/sqlserverflex/user/resource.go:420]: Same here

File: stackit/internal/services/sqlserverflex/user/resource.go (hunk 1 above)
Comment (GokceGK) [reply stackit/internal/services/sqlserverflex/user/resource.go:370]: in the response of the api, the field is named as default_database. Service team will fix it. Until then, we will leave this field out.

File: stackit/internal/services/sqlserverflex/user/resource.go [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+			"username": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"roles": schema.SetAttribute{
+				ElementType: types.StringType,
+				Required:    true,
```
Comment (joaopalet) [inline stackit/internal/services/sqlserverflex/user/resource.go:165]: Is this field really required by the API? I think we can create users without setting it

Comment (GokceGK) [reply stackit/internal/services/sqlserverflex/user/resource.go]: Roles field is now Optional

**Review by joaopalet — APPROVED** [review]

**Pull request merged**